### PR TITLE
Filter sort

### PR DESCRIPTION
**Pull request by BelideNikhil**
### Added

+ Sort by date and Priority
+ Filter by Priority and Tags
+ Dynamic Sidebar

### CODE REVIEW

**Review by poornima-alugubelly — APPROVED** [review]
everything working as expected

File: src/Components/NoteInput/NoteInput.jsx [hunk 1]
```diff
@@ -21,34 +29,48 @@ export function NoteInput() {
...
+    function resetFunction() {
+        setInputData(initialData);
+        falseStatesSetter();
+        tagsDispatchFunction({ type: "RESET_GLOBAL_TAG_STATES" });
+    }
+    function setCurrentFieldhandler(e, setterFunction) {
+        e.stopPropagation();
+        falseStatesSetter();
+        setterFunction(true);
+    }
```
Comment (msdeshmukh009) [inline src/Components/NoteInput/NoteInput.jsx:49]: Not mandatory but extracting these functions in the hook will increase the overall readability of the code.

File: src/Components/Filter/Filter.css [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    border: 1px solid var(--clr-grey-light);
+    border-radius: var(--border-radius);
+    padding: var(--padding-md);
+    box-shadow: var(--box-shadow-light);
+    background-color: var(--clr-whitesmoke);
+    width: max-content;
+    position: absolute;
+    top: 4rem;
+    left: -5.5rem;
+    z-index: 5;
```
Comment (msdeshmukh009) [inline src/Components/Filter/Filter.css:11]: Utility for z-index?

**Review by msdeshmukh009 — APPROVED** [review]
The code looks clean and UI is working as expected.

File: src/Components/Filter/Filter.css (hunk 2 above)
Comment (BelideNikhil) [reply src/Components/Filter/Filter.css:11]: Will be resolving in the next PR, thanks for pointing it

**Pull request merged**